modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/rolesUtils.ts

```diff
@@ -36,21 +38,47 @@ export async function getEmailNotificationRoles(baseResourceURL: string) {
 	const rolesResponse = (await response.json()) as RolesGroup;
 	const roles = [] as MultiSelectItem[];
 
-	(Object.entries(rolesResponse) as [keyof RolesGroup, Role[]][]).forEach(
-		([roleGroupKey, roleValues]) => {
-			roles.push({
-				children: roleValues.map(({label, name}) => {
-					return {
-						checked: false,
-						label,
-						value: name,
-					};
-				}),
-				label: roleGroupLabels[roleGroupKey],
-				value: roleGroupKey,
-			});
-		}
-	);
+	(Object.entries(rolesResponse) as [
+		keyof RolesGroup,
+		EmailNotificationRole[]
+	][]).forEach(([roleGroupKey, roleValues]) => {
+		roles.push({
+			children: roleValues.map(({label, name}) => {
+				return {
+					checked: false,
+					label,
+					value: name,
+				};
+			}),
+			label: roleGroupLabels[roleGroupKey],
+			value: roleGroupKey,
+		});
+	});
+
+	return roles;
+}
+
+export function getUserNotificationRoles(
+	rolesItems: Role[],
+	recipients: {roleName: string}[]
+) {
+	const roles = {
+		children: rolesItems
+			.filter(({name}) => name !== 'Guest')
+			.map(({externalReferenceCode, name}) => {
+				const selectedRole = !!(recipients).find(
+					(recipient) => recipient.roleName === externalReferenceCode
+				);
```

Comment:
Hey @rijema, those parentheses around `recipients` are unnecessary, you can also destructure the `recipient` to directly get the `roleName`:

```ts
const selectedRole = !!recipients.find(
	({roleName}) => roleName === externalReferenceCode
);
```